File: apps/trustim-investigation/workbench/src/types/alert.ts
/** Alert lifecycle statuses */
export type AlertStatus = 'new' | 'investigating' | 'resolved' | 'dismissed'

/** Alert severity levels */
export type AlertSeverity = 'critical' | 'high' | 'medium' | 'low' | 'info'

/** Where the alert came from */
export type AlertSource = 'inresponse' | 'iris' | 'manual' | 'playbook' | 'external'

/** An extracted IOC from an alert */
export interface AlertIOC {
  type: 'ip' | 'domain' | 'member_id' | 'email' | 'device_hash' | 'user_agent' | 'other'
  value: string
}

/** Core alert entity */
export interface Alert {
  id: string
  /** External ID from InResponse or IRIS */
  external_id?: string
  title: string
  description: string
  status: AlertStatus
  severity: AlertSeverity
  source: AlertSource

  /** Alert type/category (maps to investigation-router categories) */
  alert_type: string
  /** Assignee (LinkedIn username) */
  assignee?: string

  created_at: string
  updated_at: string
  resolved_at?: string

  /** Linked investigation session IDs */
  session_ids: string[]
  /** Related alert IDs (same IOCs, same incident, etc.) */
  related_alert_ids: string[]
  /** Extracted IOCs for cross-alert correlation */
  iocs: AlertIOC[]

  /** Suggested playbook ID based on alert type */
  suggested_playbook_id?: string
  /** InResponse incident ID if linked */
  incident_id?: string

  /** Arbitrary metadata from the source system */
  metadata: Record<string, unknown>
  /** Tags for filtering */
  tags: string[]
}

/** Lightweight alert summary for list views */
export interface AlertSummary {
  id: string
  external_id?: string
  title: string
  status: AlertStatus
  severity: AlertSeverity
  source: AlertSource
  alert_type: string
  assignee?: string
  created_at: string
  updated_at: string
  session_count: number
  ioc_count: number
  related_count: number
  tags: string[]
}

/** Filters for the alert queue */
export interface AlertFilters {
  status?: AlertStatus[]
  severity?: AlertSeverity[]
  source?: AlertSource[]
  alert_type?: string
  assignee?: string
  search?: string
  date_from?: string
  date_to?: string
}

/** Create a new empty alert */
export function createAlert(overrides: Partial<Alert> & { id: string; title: string }): Alert {
  return {
    description: '',
    status: 'new',
    severity: 'medium',
    source: 'manual',
    alert_type: '',
    created_at: new Date().toISOString(),
    updated_at: new Date().toISOString(),
    session_ids: [],
    related_alert_ids: [],
    iocs: [],
    metadata: {},
    tags: [],
    ...overrides,
  }
}

File: apps/trustim-investigation/workbench/server-py/app/models/alert.py
from __future__ import annotations

from typing import Literal

from pydantic import BaseModel, Field


AlertStatus = Literal["new", "investigating", "resolved", "dismissed"]
AlertSeverity = Literal["critical", "high", "medium", "low", "info"]
AlertSource = Literal["inresponse", "iris", "manual", "playbook", "external"]
IOCType = Literal["ip", "domain", "member_id", "email", "device_hash", "user_agent", "other"]


class AlertIOC(BaseModel):
    type: IOCType
    value: str


class Alert(BaseModel):
    model_config = {"extra": "allow"}

    id: str
    external_id: str | None = None
    title: str
    description: str = ""
    status: AlertStatus = "new"
    severity: AlertSeverity = "medium"
    source: AlertSource = "manual"
    alert_type: str = ""
    assignee: str | None = None
    created_at: str = ""
    updated_at: str = ""
    resolved_at: str | None = None
    session_ids: list[str] = Field(default_factory=list)
    related_alert_ids: list[str] = Field(default_factory=list)
    iocs: list[AlertIOC] = Field(default_factory=list)
    suggested_playbook_id: str | None = None
    incident_id: str | None = None
    metadata: dict = Field(default_factory=dict)
    tags: list[str] = Field(default_factory=list)


class AlertSummary(BaseModel):
    id: str
    external_id: str | None = None
    title: str
    status: AlertStatus
    severity: AlertSeverity
    source: AlertSource
    alert_type: str = ""
    assignee: str | None = None
    created_at: str = ""
    updated_at: str = ""
    session_count: int = 0
    ioc_count: int = 0
    related_count: int = 0
    tags: list[str] = Field(default_factory=list)


class AlertFilters(BaseModel):
    status: list[AlertStatus] | None = None
    severity: list[AlertSeverity] | None = None
    source: list[AlertSource] | None = None
    alert_type: str | None = None
    assignee: str | None = None
    search: str | None = None
    date_from: str | None = None
    date_to: str | None = None
